src/features/posts.ts

```diff
@@ -0,0 +1,49 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+export interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchPosts = createAsyncThunk('posts/fetch', (userId: number) => {
+  return getUserPosts(userId);
```

Comment:
Ensure that the `getUserPosts` function is correctly implemented and handles errors properly. This will help prevent issues when fetching posts.